include/deal.II/grid/grid_tools.h

```diff
@@ -974,6 +974,41 @@ namespace GridTools
   get_patch_around_cell(const typename MeshType::active_cell_iterator &cell);
 
 
+  /**
+   * This function runs through the degrees of freedom defined by the
+   * Container and for each dof constructs a vector of active_cell_iterators
+   * representing the cells of support of the nodal basis element
+   * at that degree of freedom. This function was designed for the
+   * implementation of local projections, for instance the Clement interpolant,
+   * in conjunction with other local patch functions like
+   * GridTools::build_triangulation_from_patch.
+   *
+   * Containers built on top of Triangulation<dim> or
+   * parallel:distributed::Triangulation<dim> are supported and handled
+   * appropriately.
+   *
+   * It is necessary that the finite element underlying the Container used has
+   * degrees of freedom on faces (2d or 3d) and lines (in 3d).  This unfortunately
+   * precludes the FE_DGQ<dim> finite element.  Likewise, the finite element
+   * must have nodal basis elements for this implementation to make sense.
```

Comment:
Likewise, here I think all that matters is whether a DoF is *logically* associated with a vertex, line, quad, or hex. You can have a modal basis where DoFs are associated with edges, and that should be totally fine.